### PR TITLE
Added graceful handling of errors related to peak annotation file conversion

### PR DESCRIPTION
**Pull request by hepcat72**
## Summary Change Description

After this PR, when you submit the carnosine study's accucor files to the submission start page, instead of a 500 error, you get:

image

## Affected Issues/Pull Requests

- Partially addresses #1320
- Merges into main

## Review Notes

See comments in-line.

## Checklist

This pull request will be merged once the following requirements are met.  The
author and/or reviewers should uncheck any unmet requirements:

- Review requirements
  - Minimum approvals: 1 
  - No changes requested
  - All blocking issues resolved by reviewers
  - Specific reviewers: @__add_username_here__
    
  - Review period: 2 days 
- Associated issue/pull request requirements:
  
  - [x] All requirements in affected issues marked "resolved" are satisfied
  - [x] All required pull requests are merged *(or none)*
- Basic requirements
  
  - [x] [All linters pass](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#linting)
  - [x] [All tests pass](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#quality-control)
  - [x] All conflicts resolved
- Overhead requirements
  
  - [x] [New/changed method tests implemented/updated *(or no change)*](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#test-implementation)
  - [ ] [Updated *Unreleased* section of `changelog.md` *(or no change)*](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/changelog.md)
  - [x] [Migrations created & committed *(or no change)*](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#migration-process)

### CODE REVIEW

File: DataRepo/loaders/peak_annotations_loader.py [hunk 1]
```diff
@@ -356,9 +356,18 @@ def __init__(self, *args, multrep_suggestion=None, **kwargs):
             self.peakgroupconflicts.get_selected_representations()
```
Comment (hepcat72) [inline DataRepo/loaders/peak_annotations_loader.py:356]: C. All the changes in this and the next file simply wrap the loader calls in `try` blocks.

File: DataRepo/views/upload/submission.py [hunk 2]
```diff
@@ -529,15 +529,22 @@ def set_files(
...
+                                filename=peak_annot_filename,
+                                # We don't need the default sequence info - we only want to read the file
+                            )
                         )
-                    )
+                    except AggregatedErrors as aes:
+                        self.load_status_data.set_load_exception(
+                            aes,
+                            peak_annot_filename,
+                        )
```
Comment (hepcat72) [inline DataRepo/views/upload/submission.py:546]: C. This buffers the exceptions into the `MultiLoadStatus` object that is presented to the user on the results page.

File: DataRepo/views/upload/submission.py [hunk 3]
```diff
@@ -529,15 +529,22 @@ def set_files(
...
+                        self.peak_annotations_loaders.append(None)
```
Comment (hepcat72) [inline DataRepo/views/upload/submission.py:547]: C. Loader instantiation failed, so we add a `None` to the list of loaders.

File: DataRepo/views/upload/submission.py [hunk 4]
```diff
@@ -610,12 +624,11 @@ def set_files(
...
-                    for pal_exc in pal.aggregated_errors_object.exceptions:
-                        self.load_status_data.set_load_exception(
-                            pal_exc,
-                            peak_annot_filename,
-                        )
+                if pal is not None and len(pal.aggregated_errors_object.exceptions) > 0:
+                    self.load_status_data.set_load_exception(
+                        pal.aggregated_errors_object,
+                        peak_annot_filename,
+                    )
```
Comment (hepcat72) [inline DataRepo/views/upload/submission.py:631]: C. After having reviewed the `set_load_exception` code, I realized that the loop was unnecessary.

**Review by narzouni — APPROVED** [review]
LGTM. This PR adds try blocks for handling exceptions for file conversions.

**Pull request merged**